dbms/mysql/src/cloud/benchflow/collectors/mysql.go

```diff
@@ -1,90 +1,80 @@
 package main
  
 import (
-    "fmt"
     "net/http"
     "os"
     "os/exec"
     "github.com/benchflow/commons/minio"
     "github.com/benchflow/commons/kafka"
-    "log"
     "strings"
+    "encoding/json"
 )
 
+type Response struct {
+  Successful bool
+}
+
 func backupHandler(w http.ResponseWriter, r *http.Request) {
 	if r.Method != "PUT" {
 		w.WriteHeader(405)
 		return	
 	}
 	// Generating key for Minio
-	//databaseMinioKey := minio.GenerateKey(os.Getenv("MYSQL_DB_NAME"))
 	databaseMinioKey := minio.GenerateKey(os.Getenv("MYSQL_DB_NAME"), os.Getenv("BENCHFLOW_TRIAL_ID"), os.Getenv("BENCHFLOW_EXPERIMENT_ID"), os.Getenv("BENCHFLOW_CONTAINER_NAME"), os.Getenv("BENCHFLOW_COLLECTOR_NAME"), os.Getenv("BENCHFLOW_DATA_NAME"))
-
-	log.Printf("Minio Key: " + databaseMinioKey)
+	
+	// Save whole database as mysqldump
+	cmd := exec.Command("mysqldump", "-h", os.Getenv("MYSQL_HOST"), "-P", os.Getenv("MYSQL_PORT"), "-u", os.Getenv("MYSQL_USER"), "-p" + os.Getenv("MYSQL_USER_PASSWORD"), os.Getenv("MYSQL_DB_NAME"))
+	outfile, err := os.Create("/app/"+os.Getenv("MYSQL_DB_NAME")+"_backup.sql")
+    if err != nil {
+        http.Error(w, err.Error(), http.StatusInternalServerError)
+	    return
+    }
+    cmd.Stdout = outfile
+    cmd.Start()
+    err = cmd.Wait()
+    if err != nil {
+        http.Error(w, err.Error(), http.StatusInternalServerError)
+	    return
+    }
+    outfile.Close()
+   
+    minio.GzipFile("/app/"+os.Getenv("MYSQL_DB_NAME")+"_backup.sql")
+    minio.SendGzipToMinio("/app/"+os.Getenv("MYSQL_DB_NAME")+"_backup.sql.gz", os.Getenv("MINIO_HOST"), os.Getenv("MINIO_PORT"), databaseMinioKey+"/"+os.Getenv("MYSQL_DB_NAME")+"_backup.sql", os.Getenv("MINIO_ACCESSKEYID"), os.Getenv("MINIO_SECRETACCESSKEY"))
+   
+	err = os.Remove("/app/"+os.Getenv("MYSQL_DB_NAME")+"_backup.sql.gz")
```

Comment:
@VincenzoFerme When you gzip something it overwirtes the uncompressed file, so I only have to remove the gzipped file because that's all there is.